### PR TITLE
[GLUTEN-2038][VL] Improve parquet write using arrow chunked array

### PR DESCRIPTION
**Pull request by ulysses-you**
see https://github.com/oap-project/gluten/pull/2039

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @JkSelf @zhouyuan thank you

File: velox/dwio/parquet/writer/Writer.h [hunk 1]
```diff
@@ -81,18 +81,19 @@ class DataBufferSink : public arrow::io::OutputStream {
...
   // Constructts a writer with output to 'sink'. A new row group is
-  // started every 'rowsInRowGroup' top level rows. 'pool' is used for
+  // started every 'maxRowGroupBytes' top level rows. 'pool' is used for
   // temporary memory. 'properties' specifies Parquet-specific
   // options.
   Writer(
       std::unique_ptr<dwio::common::DataSink> sink,
       memory::MemoryPool& pool,
-      int32_t rowsInRowGroup,
+      int64_t maxRowGroupBytes,
```
Comment (ulysses-you) [inline velox/dwio/parquet/writer/Writer.h:90]: seems `rowsInRowGroup` is dead code

**Review by JkSelf — APPROVED** [review]
LGTM. And can you help to upstream this optimization into upstream?

**Comment by ulysses-you** [comment]
@JkSelf sure

**Pull request merged**